students/WesleyM/Lesson9/mailroom.py

```diff
@@ -0,0 +1,120 @@
+class donor:
+    def __init__(self, name, donation = None):
+        if not donation:
+            donation = []
+        self.name = name
+        self.donation = donation
+        self.total = sum(self.donation)
+        if donation:
+            self.average = self.total/len(self.donation)
+        else:
+            self.average = 0
+    def add_donation(self, amount):
+        self.donation.append(amount)
+        self.total += amount
+        self.average = self.total / len(self.donation)
+
+class DonorDatabase:
+
+    def __init__(self, name):
+        if not name:
+            self.donors = []
+        else:
+            self.donors = name
+    
+    def add_donor(self, name):
+        self.donors.append(name)
+        
+    def get_donor(self, name):
+        for d in self.donors:
+            if d.name == name:
+                return d
+
+    def get_all_donor_names(self):
+        return [donor.name for donor in self.donors]
+        
+    def sort_donors(self):
+        return sorted(self.get_all_donor_names())
+    
+    def donor_input(self):
+        return input("Enter a donor name or input 'List' for a list of donors\n>")
+    
+    def send_thankyou(self):
+        don_input = None
+        while not don_input:
+            don_input = self.donor_input()
+            if don_input.lower() == "list":
+                print(self.sort_donors())
+                don_input = None
+
+        donation = None
+        while not donation:
+            try:
+                donation = self.donation_prompt()
+            except ValueError:
+                print("Enter donations numerically")
+
+        if don_input not in self.get_all_donor_names():
+            try:
+                d1 = donor(don_input, [donation])
+                self.add_donor(d1)
+            except ValueError:
+                print("Please enter a name")
+        else:
+            for d in self.donors:
+                if d.name == don_input:
+                    d.add_donation(donation)
+
+        print("Thank you {} for your donation of ${:.2f}".format(d1.name, d1.total))
```

Comment:
This breaks for existing donors, since you only assign the `d1` variable if it's a new donor